pkg/suggestion/v1beta1/hyperopt/base_service.py

```diff
@@ -62,14 +64,75 @@ def create_hyperopt_domain(self):
         # hyperopt.hp.uniform('x2', -10, 10)}
         hyperopt_search_space = {}
         for param in self.search_space.params:
-            if param.type == INTEGER:
-                hyperopt_search_space[param.name] = hyperopt.hp.quniform(
-                    param.name, float(param.min), float(param.max), float(param.step)
-                )
-            elif param.type == DOUBLE:
-                hyperopt_search_space[param.name] = hyperopt.hp.uniform(
-                    param.name, float(param.min), float(param.max)
-                )
+            if param.type in [INTEGER, DOUBLE]:
+                if param.distribution == api_pb2.UNIFORM or param.distribution is None:
+                    if param.step:
+                        hyperopt_search_space[param.name] = hyperopt.hp.quniform(
+                            param.name,
+                            float(param.min),
+                            float(param.max),
+                            float(param.step),
+                        )
+                    else:
+                        if param.type == INTEGER:
+                            hyperopt_search_space[param.name] = hyperopt.hp.uniformint(
+                                param.name, float(param.min), float(param.max)
+                            )
+                        else:
+                            hyperopt_search_space[param.name] = hyperopt.hp.uniform(
+                                param.name, float(param.min), float(param.max)
+                            )
+                elif param.distribution == api_pb2.LOG_UNIFORM:
+                    if param.step:
+                        hyperopt_search_space[param.name] = hyperopt.hp.qloguniform(
+                            param.name,
+                            math.log(float(param.min)),
+                            math.log(float(param.max)),
+                            float(param.step),
+                        )
+                    else:
+                        hyperopt_search_space[param.name] = hyperopt.hp.loguniform(
+                            param.name,
+                            math.log(float(param.min)),
+                            math.log(float(param.max)),
```

Comment:
Do we need to take logarithm from the values before passing them to the `loguniform` ?

Reply:
We convert `param.min` and `param.max` to logarithmic scale as when optimizing, the sampled value is constrained to the interval `[exp(low), exp(high)]`. Refer to `hp.loguniform` documentation [here](http://hyperopt.github.io/hyperopt/getting-started/search_spaces/#parameter-expressions).

Also [here](https://github.com/facebookresearch/nevergrad/blob/c25cfd7b0aa72749331754ba5a20b67fa2615dc7/nevergrad/optimization/externalbo.py#L66-L68) in the `nevergrad` implementation the upper and lower bounds were converted to logarithmic scale for `hp.loguniform`

@andreyvelich @tenzen-y